client/src/pages/ViewRequestStatus/RequestStatusView.js

```diff
@@ -1,16 +1,40 @@
 import React, { useEffect, useState } from "react";
+import io from "socket.io-client";
 // import { Navigate } from "react-router-dom";
 
 import { useParams } from "react-router-dom";
 import ButtonComponent from "./ButtonComponent";
+const socket = io(window.origin, { path: "/api/socket.io" });
 
 export function RequestStatus() {
 	const [request, setRequest] = useState(null);
 	const [donation, setDonation] = useState(null);
 	const [needsReloading, setNeedsReloading] = useState(false);
 
+
 	// this helps get the id from the router
 	const { id } = useParams();
+	useEffect(() => {
+		socket.on("connect", () => {
+			console.log("conncted");
+		});
+		socket.on("laptop_request:statusChanged", ({ laptopRequestId }) => {
+			setNeedsReloading((previousNeedsReloading) => {
+				return !previousNeedsReloading;
+			});
+			if (request != null && request.status == "WAITING") {
```

Comment:
at this point, the request value can't really be relied on (and i would force you to use the request as a useEffect dependency which we don't want)

Reply:
(so we don't need this if block)